.github/workflows/tox-checkbox.yaml

```
@@ -0,0 +1,152 @@
+name: Tox and push to codecov core components
+permissions:
+  contents: read
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - checkbox-ng/**
+      - checkbox-support/**
+      - providers/base/**
+      - providers/certification-client/**
+      - providers/certification-server/**
+      - providers/genio/**
+      - providers/gpgpu/**
+      - providers/resource/**
+      - providers/sru/**
+      - providers/genio/**
+      - providers/iiotg/**
+      - .github/workflows/tox-checkbox.yaml
+  pull_request:
+    branches: [ main ]
+    paths:
+      - checkbox-ng/**
+      - checkbox-support/**
+      - providers/base/**
+      - providers/certification-client/**
+      - providers/certification-server/**
+      - providers/gpgpu/**
+      - providers/resource/**
+      - providers/sru/**
+      - providers/genio/**
+      - providers/iiotg/**
+      - .github/workflows/tox-checkbox.yaml
+  workflow_dispatch:
+
+jobs:
+  get_path_matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      path_flag: ${{ steps.paths.outputs.paths }}
+    steps:
+      - name: Checkout Checkbox monorepo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - id: paths
+        run: |
+          CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+          if [ "$CURRENT_BRANCH" == "main" ]; then
+            echo "Workflow triggered on main, diffing with HEAD~1"
+            DIFF=$(git diff --name-only HEAD~1)
+          else
+            echo "Workflow triggered on a branch, diffing with origin/main"
+            DIFF=$(git diff --name-only origin/main)
+          fi
+          changed(){ echo "$DIFF" | grep -o $1; }
+          if changed ".github/workflows/tox-checkbox.yaml"; then
+            # when changing the workflow, re-test all tox to check they still work
+            changed(){ true; }
+          fi;
+
+          # providers have the flag provider-provider-name
+          get_codecov_flag(){ echo "$1" | sed 's|providers/|provider-|g'; }
+
+          echo -n "paths=[" >> to_output
+          for path in \
+              checkbox-ng \
+              checkbox-support \
+              providers/base \
+              providers/certification-client \
+              providers/certification-server \
+              providers/gpgpu \
+              providers/resource \
+              providers/genio \
+              providers/iiotg \
+              providers/sru; do
+            if changed $path; then
+              echo -n '{ "path":' >> to_output
```

You are already printing it to the general output, but I would also add a debug message here. Up to you, but if you do it, add a quiet flag to the grep in  `changed()` :
```suggestion
              echo "Including tox for $path" 
              echo -n '{ "path":' >> to_output
```